dss/api/bundles/__init__.py

```diff
@@ -238,7 +238,7 @@ def delete(uuid: str, replica: str, json_request_body: dict, version: str = None
         response_body = dict()  # type: dict
     else:
         status_code = requests.codes.not_found
-        response_body = dict(title="bundle not found")
+        response_body = dict(title="bundle not found", code=status_code)
```

Comment:
`DSSException` should be raised here. See, [for example](https://github.com/HumanCellAtlas/data-store/blob/62c60d96a096ef3b2c80810ddc619c9ad23a2397/dss/api/bundles/__init__.py#L54)

`not_found` should also needs to be included in the API spec for DELETE /bundle [here](https://github.com/HumanCellAtlas/data-store/blob/62c60d96a096ef3b2c80810ddc619c9ad23a2397/dss-api.yml#L1996)